src/Lykke.Service.BlockchainCashoutPreconditionsCheck.Services/ValidationService.cs

```diff
@@ -63,28 +65,37 @@ public async Task<IEnumerable<ValidationError>> ValidateAsync(CashoutModel casho
             if (asset == null)
                 throw new ArgumentValidationException($"Asset with Id-{cashoutModel.AssetId} does not exists", "assetId");
 
-            if (string.IsNullOrEmpty(asset.BlockchainIntegrationLayerId))
-                throw new ArgumentValidationException($"Given asset Id-{cashoutModel.AssetId} is not a part of Blockchain Integration Layer", "assetId");
-
-            var blockchainClient = _blockchainApiClientProvider.Get(asset.BlockchainIntegrationLayerId);
 
             List<ValidationError> errors = new List<ValidationError>(1);
 
 
-            var isBlocked = await _blackListService.IsBlockedAsync(asset.BlockchainIntegrationLayerId,
-                cashoutModel.DestinationAddress);
-
-            if (isBlocked)
+            if (asset.Id == LykkeConstants.SolarAssetId && !SolarCoinValidation.ValidateAddress(cashoutModel.DestinationAddress))
             {
-                errors.Add(ValidationError.Create(ValidationErrorType.BlackListedAddress, "Address is in the black list"));
+                errors.Add(ValidationError.Create(ValidationErrorType.AddressIsNotValid, "Address is not valid"));
             }
 
-            if (string.IsNullOrEmpty(cashoutModel.DestinationAddress) ||
-                !await blockchainClient.IsAddressValidAsync(cashoutModel.DestinationAddress))
+            if (asset.Id != LykkeConstants.SolarAssetId)
             {
-                errors.Add(ValidationError.Create(ValidationErrorType.AddressIsNotValid, "Address is not valid"));
-            }
+                if (string.IsNullOrEmpty(asset.BlockchainIntegrationLayerId))
+                    throw new ArgumentValidationException($"Given asset Id-{cashoutModel.AssetId} is not a part of Blockchain Integration Layer", "assetId");
+
+                var blockchainClient = _blockchainApiClientProvider.Get(asset.BlockchainIntegrationLayerId);
+
 
+                var isBlocked = await _blackListService.IsBlockedAsync(asset.BlockchainIntegrationLayerId,
+                    cashoutModel.DestinationAddress);
+
+                if (isBlocked)
```

Comment:
Do black lists and other functionality is supported for SLR? 
Looks like in SLR we only use 
SolarCoinValidation.ValidateAddress

Reply:
"Do black lists and other functionality is supported for SLR?"
     Not supported because SLR do not return BlockchainIntegrationLayerId property value.  Which i need in IsBlockedAsync method. Because i do not check it. Is it need to check?

"Looks like in SLR we only use SolarCoinValidation.ValidateAddress. Is it correct?"
    Yes, SolarCoinValidation.ValidateAddress used for only SLR